tests/D2L.CodeStyle.Analyzers.Test/Specs/TypeDeclarationImmutabilityAnalyzer.cs

```diff
@@ -547,83 +619,144 @@ public sealed class AnalyzedImmutableGenericClassGivenT<[Immutable] T, U> {
-		/* TypeParameterIsNotKnownToBeImmutable(U) */ Types.SomeImmutableGenericInterfaceGivenTU</* TypeParameterIsNotKnownToBeImmutable(U) */ U /**/, T> /**/ /* MemberIsNotReadOnly(Field, m_field, AnalyzedImmutableGenericClassGivenT) */ m_field /**/;
-		readonly /* TypeParameterIsNotKnownToBeImmutable(U) */ Types.SomeImmutableGenericInterfaceGivenTU</* TypeParameterIsNotKnownToBeImmutable(U) */ U /**/, T> /**/ m_field;
-		/* TypeParameterIsNotKnownToBeImmutable(U) */ Types.SomeImmutableGenericInterfaceGivenTU</* TypeParameterIsNotKnownToBeImmutable(U) */ U /**/, T> /**/ Property { get; }
-		Types.SomeImmutableGenericInterfaceGivenTU</* TypeParameterIsNotKnownToBeImmutable(U) */ U /**/, T> Property { get { return default; } }
+		static Types.SomeImmutableGenericInterfaceRestrictingTU</* TypeParameterIsNotKnownToBeImmutable(U) */ U /**/, T> /* MemberIsNotReadOnly(Field, m_field, AnalyzedImmutableGenericClassRestrictingT) */ m_field /**/;
+		static readonly Types.SomeImmutableGenericInterfaceRestrictingTU</* TypeParameterIsNotKnownToBeImmutable(U) */ U /**/, T> m_field;
+		Types.SomeImmutableGenericInterfaceRestrictingTU</* TypeParameterIsNotKnownToBeImmutable(U) */ U /**/, T> /* MemberIsNotReadOnly(Field, m_field, AnalyzedImmutableGenericClassRestrictingT) */ m_field /**/;
+		readonly Types.SomeImmutableGenericInterfaceRestrictingTU</* TypeParameterIsNotKnownToBeImmutable(U) */ U /**/, T> m_field;
+		Types.SomeImmutableGenericInterfaceRestrictingTU</* TypeParameterIsNotKnownToBeImmutable(U) */ U /**/, T> Property { get; }
+		Types.SomeImmutableGenericInterfaceRestrictingTU</* TypeParameterIsNotKnownToBeImmutable(U) */ U /**/, T> Property { get { return default; } }
 
 
 		void Method() {
 			Types.SomeGenericMethod<T, U>();
 			Types.SomeGenericMethod<U, T>();
 			Types.SomeGenericMethod<T, T>();
 			Types.SomeGenericMethod<U, U>();
-			Types.SomeGenericMethodGivenT<T, U>();
-			Types.SomeGenericMethodGivenT</* TypeParameterIsNotKnownToBeImmutable(U) */ U /**/, T>();
-			Types.SomeGenericMethodGivenT<T, T>();
-			Types.SomeGenericMethodGivenT </* TypeParameterIsNotKnownToBeImmutable(U) */ U /**/, U>();
-			Types.SomeGenericMethodGivenU<T, /* TypeParameterIsNotKnownToBeImmutable(U) */ U /**/>();
-			Types.SomeGenericMethodGivenU<U, T>();
-			Types.SomeGenericMethodGivenU<T, T>();
-			Types.SomeGenericMethodGivenU<U, /* TypeParameterIsNotKnownToBeImmutable(U) */ U /**/>();
-			Types.SomeGenericMethodGivenTU<T, /* TypeParameterIsNotKnownToBeImmutable(U) */ U /**/>();
-			Types.SomeGenericMethodGivenTU</* TypeParameterIsNotKnownToBeImmutable(U) */ U /**/, T>();
-			Types.SomeGenericMethodGivenTU<T, T>();
-			Types.SomeGenericMethodGivenTU</* TypeParameterIsNotKnownToBeImmutable(U) */ U /**/, /* TypeParameterIsNotKnownToBeImmutable(U) */ U /**/>();
+			Types.SomeGenericMethodRestrictingT<T, U>();

```

Comment:
Methods were renamed to `Restricting` since it better reflects what `[Immutable]` does on a type parameter